### PR TITLE
Remove cgmath

### PR DESCRIPTION
**Pull request by cwfitzgerald**
This PR completely removes cgmath from bve-reborn, fully replacing it with either glam for floats and a simple hand rolled impl for integers.

Additional changes:
- Minor c api fixes
- Removes build caches because they're lame

### CODE REVIEW

File: bve-derive/src/serde_proxy.rs [hunk 1]
```diff
@@ -191,23 +187,11 @@ fn generate_proxy_object(name: &Ident, fields: &[Field]) -> TokenStream2 {
...
-                    #x_attribute
-                    #x_new: #x_inner,
-                };
-                let from_fields = quote! {
-                    #original_name: ::cgmath::Vector1::new(#x_conversion),
-                };
-                (proxy_fields, from_fields)
-            }
-            vec if vec.last().map(String::as_str) == Some("Vector2") => {
+            vec if match vec.last().map(String::as_str) {
```
Comment (DethRaid) [inline bve-derive/src/serde_proxy.rs:190]: Where does `vec` come from?

File: bve-derive/src/serde_proxy.rs [hunk 2]
```diff
@@ -191,23 +187,11 @@ fn generate_proxy_object(name: &Ident, fields: &[Field]) -> TokenStream2 {
...
+                Some("Vec2") | Some("BVec2") | Some("UVec2") | Some("IVec2") => true,
```
Comment (DethRaid) [inline bve-derive/src/serde_proxy.rs:191]: Is matching string type names really the best way to do this?

File: bve-native/include/bve_prefix.h [hunk 3]
```diff
@@ -9,3 +9,22 @@
...
+typedef struct {
+    float x;
+    float y;
+} BVE_Vec2;
+
+typedef struct {
+    float x;
+    float y;
+    float z;
+    float _z;
```
Comment (DethRaid) [inline bve-native/include/bve_prefix.h:22]: I thought you renamed this to `_padding`?

File: bve-derive/src/serde_proxy.rs (hunk 2 above)
Comment (cwfitzgerald) [reply bve-derive/src/serde_proxy.rs:191]: No, but holy fuck would doing it properly be hard. It's a hacky mess.

File: bve-derive/src/serde_proxy.rs (hunk 1 above)
Comment (cwfitzgerald) [reply bve-derive/src/serde_proxy.rs:190]:
```rust
match thing {
    other_binding_to_thing if condition => {}
}
```
is the basic syntax

Comment (DethRaid) [reply bve-derive/src/serde_proxy.rs:190]: I figured that out... but I didn't see the declaration of `vec` in the first place

File: bve-derive/src/serde_proxy.rs (hunk 2 above)
Comment (DethRaid) [reply bve-derive/src/serde_proxy.rs:191]: I'll accept it

File: bve-native/include/bve_prefix.h (hunk 3 above)
Comment (cwfitzgerald) [reply bve-native/include/bve_prefix.h]: Visual studio doesn't autosave. I'm used to clion which does.

**Pull request merged**